src/maths/linear.cpp

```diff
@@ -0,0 +1,290 @@
+#include <cmath>
+#include "linear.hpp"
+
+#define UNUSED_PARAM(x)  (void)x
```

Comment:
We can declare this in macro.h since it could be useful for other part of the engine